src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/WebSocketsTransport.cs

```diff
@@ -390,18 +393,45 @@ private async Task ProcessSocketAsync(WebSocket socket, Uri url, bool isReconnec
                 {
                     _application.Output.CancelPendingFlush();
                 }
+
+                // Should not throw
+                await receiving.ConfigureAwait(false);
             }
         }
 
-        if (_useStatefulReconnect && !_gracefulClose)
+        var cleanup = true;
+        try
         {
-            if (!UpdateConnectionPair())
+            if (_useStatefulReconnect && !_gracefulClose)
```

Comment:
```suggestion
            if (!_gracefulClose && UpdateConnectionPair())
```

`UpdateConnectionPair()` already checks `_useStatefulReconnect` and will always return `true` in that case. Then you can get rid of the entire if, right below. GitHub wouldn't let me do it in a single comment.